File: Array & String/Remove Duplicates from Sorted Array II/RemoveDuplicates.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        var filteredArray = nums.GroupBy(num => num)
                                .SelectMany(group =>
                                {
                                    int count = group.Count();
                                    return count >= 2 ? group.Take(2) : group;
                                }).ToArray();
        
        Array.Copy(filteredArray, nums, filteredArray.Length);
        return filteredArray.Length;
    }
}

File: Array & String/Remove Duplicates from Sorted Array II/remove_duplicates.py
from typing import List


def remove_duplicates(nums: List[int]) -> int:
    if len(nums) <= 2:
        return len(nums)

    i = 2
    for j in range(i):
        if nums[j] != nums[i - 2]:
            nums[i] = nums[j]
            i += 1

    return i
